### PR TITLE
BXMSDOC-3291-master: Single-sourced authoring asset content in Drools community doc.

### PR DESCRIPTION
**Pull request by sterobin**
See [JIRA](https://issues.jboss.org/browse/BXMSDOC-3291) for details.

### CODE REVIEW

File: doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/XLSDecisionTable-section.adoc [hunk 1]
```diff
@@ -1,22 +1,248 @@
...
+then
+  list.add("Old man stilton");
+end
+
+rule "Cheese fans_2"
+when
+  Person(age == 21)
+  Cheese(type == "cheddar")
+then
+  list.add("Young man cheddar");
```
Comment (jomarko) [inline doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/XLSDecisionTable-section.adoc:228]: To be consistent with my other reviews, I think `mvel` code snippets should use whitespace arround brackets, so:
```
Person( age == 42 )
Cheese( type == "stilton" )
list.add( "Old man stilton" );
Person( age == 21 )
Cheese( type == "cheddar" )
list.add( "Young man cheddar" );
```

File: doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/XLSDecisionTable-section.adoc [hunk 2]
```diff
@@ -1,22 +1,248 @@
...
+
+[source,java]
+----
+KieSession ksession = kc.newKieSession( "TemplatesKS" );
+
+// now create some test data
+ksession.insert( new Cheese( "stilton", 42 ) );
+ksession.insert( new Person( "michael", "stilton", 42 ) );
+final List<String> list = new ArrayList<String>();
+ksession.setGlobal( "list", list );
```
Comment (jomarko) [inline doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/XLSDecisionTable-section.adoc:242]: To be consistent with my other reviews, I think `java` code snippets shouldn't use whitespace around brackets:
```java
KieSession ksession = kc.newKieSession("TemplatesKS");
ksession.insert(new Cheese("stilton", 42));
ksession.insert(new Person("michael", "stilton", 42));
final List<String> list = new ArrayList<String>();
ksession.setGlobal("list", list);
```

**Review by jomarko — COMMENTED** [review]
Comments to code snippets, looking to content in generated output now.

File: doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/guided-rules-WHEN-proc.adoc [hunk 3]
```diff
@@ -44,11 +44,11 @@ Bankruptcy(application == a ).
 +
```
Comment (jomarko) [inline doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/guided-rules-WHEN-proc.adoc]: I see the snippet `Bankruptcy(application == a ).` was not touched, however would you mind to update to `Bankruptcy( application == a ).` ?

**Comment by jomarko** [comment]
@sterobin please ignore my comment about code samples, I had a look and have prepared fix in commit https://github.com/jomarko/kie-docs/commit/0bcb367af754b0dadac4c44fbd88c0e1cbba709d, will open PR once your changes are on kiegroup/kie-docs/master. Or feel free to cherry pick.

**Comment by sterobin** [comment]
Thanks @jomarko. I'll ignore the snippet fixes then.

File: doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/TestScenarioEditor-section.adoc [hunk 4]
```diff
@@ -1,75 +1,128 @@
...
-.New Input popup
-image::Workbench/AuthoringAssets/NewInput.png[align="center"]
-
-
-* Call method on an existing fact - Call a method from a fact in the beginning of the rule execution.
+= Test scenarios
+
+Test scenarios in {PRODUCT} enable you to validate the functionality of business rules and business rule data before deploying them into a production environment. With a test scenario, you use data from your project to set given conditions and expected results based on one or more defined business rules. When you run the scenario, the expected results and actual results of the rule instance are compared. If the expected results match the actual results, the test is successful. If the expected results do not match the actual results, then the test fails.
+
+Test scenarios can be executed one at the time or as a group. The group execution contains all the scenarios from one package. Test scenarios are independent, so one scenario cannot affect or modify the other. You can run test scenarios at any time during project development in {CENTRAL}. You do not have to compile or deploy your decision service to run test scenarios.
```
Comment (manstis) [inline doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/TestScenarioEditor-section.adoc]: "..one at **a** time or..."?

File: doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/TestScenarioEditor-section.adoc [hunk 5]
```diff
@@ -1,75 +1,128 @@
...
+
+The *GIVEN* section defines input facts for the test. For example, if an `Underage` rule in the project declines loan applications for applicants under the age of 21, then the *GIVEN* facts in the test scenario could be `Applicant` with `age` set to some integer less than 21.
+
+.Prerequisite
+All data objects required for your test scenario have been created or imported and are listed in the *Data Objects* tab of the test scenarios designer.
+
+.Procedure
+. In the test scenarios designer, click *GIVEN* to open the *New input* window with the available facts.
++
+.Add GIVEN input to the test scenario
```
Comment (manstis) [inline doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/TestScenarioEditor-section.adoc:71]: Would it not make more sense to document the "Add GIVEN..." _before_ "Add EXPECTED..." section?

**Review by manstis — COMMENTED** [review]
@sterobin I agree with @jomarko that the http://file.rdu.redhat.com/~sterobin/BXMSDOC-3291_DROOLS/#_rule_templates (19.7.6) is out of place under "Authoring Assets" as what is documented cannot be accomplished within the Workbench.

I think its inclusion under Section 6 ("Rule Language Reference", perhaps 6.11) would be better. Taking DSLs as an example; it is included in both Section 6 (where technical details are presented) and Section 19 (where use within the Workbench is covered).

Otherwise just a couple of minor comments.

File: doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/TestScenarioEditor-section.adoc (hunk 4 above)
Comment (sterobin) [reply doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/TestScenarioEditor-section.adoc]: Oops! Good eye.

File: doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/TestScenarioEditor-section.adoc (hunk 5 above)
Comment (sterobin) [reply doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/TestScenarioEditor-section.adoc:71]: Thanks! I somehow switched these in this PR (when I copied and pasted). It's correct elsewhere. Fixed now. Sorry about that!

**Comment by sterobin** [comment]
@manstis and @jomarko, I followed up with Mike about the Rule templates thing in the Decision tables chapter and we decided to actually leave it there. Reason is that this section is very much in the context of decision tables, and used to be in the DT section in chap. 6 that I consolidated with the rule-authoring chap. The fact that rule templates is not a workbench thing is, imo, not relevant because neither are XLS decision tables :) But more importantly, that rule-authoring chapter to me is not about the workbench, but about rule-authoring methods, the majority of which logically reside in the workbench.

So, we're leaving it there for now, if that's okay.

**Pull request merged**